Source/DafnyDriver/DafnyDriver.cs

```diff
@@ -756,7 +763,9 @@ private record TargetPaths(string Directory, string Filename) {
 
     static void WriteDafnyProgramToFiles(DafnyOptions options, TargetPaths paths, bool targetProgramHasErrors, string targetProgramText,
       string/*?*/ callToMain, Dictionary<string, string> otherFiles, TextWriter outputWriter) {
-      WriteFile(paths.Filename, targetProgramText, callToMain);
+      if (targetProgramText.Length != 0) {
```

Comment:
The backend has no control over that unfortunately, the best it can do is not write anything to the `ConcreteSyntaxTree` its handed by the `Compile` method.